fe/fe-core/src/main/java/org/apache/doris/catalog/Tablet.java

```diff
@@ -691,6 +691,23 @@ public TabletStatus getColocateHealthStatus(long visibleVersion,
         return TabletStatus.HEALTHY;
     }
 
+    public Set<Long> getUnavailableBeIdsForColocate(Set<Long> backendsSet, long visibleVersion) {
+        Set<Long> replicaBackendIds = getBackendIds();
+        Set<Long> unavailableBeIds = Sets.newHashSet(Sets.difference(backendsSet, replicaBackendIds));
+        for (Replica replica : replicas) {
+            long backendId = replica.getBackendId();
+            if (backendsSet.contains(backendId)) {
+                if (!replica.isAlive()) {
+                    unavailableBeIds.add(backendId);
+                }
+                if (replica.getLastFailedVersion() > 0 || replica.getVersion() < visibleVersion) {
```

Comment:
replica missing version is very common,  it can clone version from other BE.
do not check their version.